scripts/core/helpers/network.tsx

```diff
@@ -173,3 +174,30 @@ export function uploadFileWithProgress<T>(
             });
         });
 }
+
+export function trackArticleActionProgress<T>(
+    p: () => Promise<T>,
+    itemId: string,
+    successMessage: string,
+    errorMessage: string,
+): Promise<T> {
+    dispatchEvent(new CustomEvent('action-loading', {
```

Comment:
would be good to call it `article-action-loading` so if anyone is logging events they would discover quicker from which part is it coming from.